cmd/internal/providers.go

```diff
@@ -82,10 +82,23 @@ func CreateProviders(ctx context.Context, logger *slog.Logger, gcpProjects, awsP
 	return providers, nil
 }
 
+// ProviderConfig holds the variables used by ProviderFlags
+type ProviderConfig struct {
+	GCPProjects       StringSliceFlag
+	AWSProfiles       StringSliceFlag
+	AzureSubs         StringSliceFlag
+	GCPProviderName   string
+	AWSProviderName   string
+	AzureProviderName string
+}
+
 // ProviderFlags adds the provider configuration flags to the given
 // flag set.
-func ProviderFlags(fs *flag.FlagSet, gcp, aws, azure *StringSliceFlag) {
-	fs.Var(gcp, "gcp.project", "GCP project ID (can be specified multiple times)")
-	fs.Var(aws, "aws.profile", "AWS profile (can be specified multiple times)")
-	fs.Var(azure, "azure.sub", "Azure subscription (can be specified multiple times)")
+func ProviderFlags(fs *flag.FlagSet, pc *ProviderConfig) {
+	fs.Var(&pc.GCPProjects, "gcp.project", "GCP project ID (can be specified multiple times)")
+	fs.Var(&pc.AWSProfiles, "aws.profile", "AWS profile (can be specified multiple times)")
+	fs.Var(&pc.AzureSubs, "azure.sub", "Azure subscription (can be specified multiple times)")
+	fs.StringVar(&pc.GCPProviderName, "gcp.providername", gcp.DefaultProviderName, `GCP provider name to use, default: "GCP" (e.g. "GKE")`)
+	fs.StringVar(&pc.AWSProviderName, "aws.providername", aws.DefaultProviderName, `AWS provider name to use, default: "AWS" (e.g. "EKS")`)
+	fs.StringVar(&pc.AzureProviderName, "azure.providername", azure.DefaultProviderName, `Azure provider name to use, default: "Azure" (e.g. "AKS")`)
```

Comment:
Let's revert the IDs flags as they were, and instead do this in the for the package variables we would be introducing:
```suggestion
	fs.StringVar(&gcp.ProviderName, "gcp.providername", gcp.DefaultProviderName, `GCP provider name to use, default: "GCP" (e.g. "GKE")`)
	fs.StringVar(&aws.ProviderName, "aws.providername", aws.DefaultProviderName, `AWS provider name to use, default: "AWS" (e.g. "EKS")`)
	fs.StringVar(&azure.ProviderName, "azure.providername", azure.DefaultProviderName, `Azure provider name to use, default: "Azure" (e.g. "AKS")`)
```